autoinstall/README.md

```diff
@@ -0,0 +1,16 @@
+# Autoinstall
+By using autoinstall we could automate the installer like what `presed` is used for `d-i`.
+
+## Usage
+1. Upload your `user-data` to a www server. You may use `basic.yaml` as an example. Rename `basic.yaml` and put it 
+   into a www server to serve it.
+2. `touch <your www server>/meta-data` to create an empty but necessary file required by `cloud-init`.
+3. Edit `grub.cfg` to include the necessary kernel parameters `autoinstall "ds=nocloud-net;s=http://<www-server-url>/"`. For example,
```

Comment:
`grub.cfg` will be UEFI-specific. `pxelinux` users should also be able to use it by changing their `APPEND` line in the same way, so might be good to make this generic. In fact, have you considered adding an `--autoinstall-url` option to the script to do it for you? We already have the `--extra-args` parameter which modifies the command line in both config files. Difference here is that it'd need to be inserted before the `---`

Reply:
My gut feeling shows this is an good idea. Let me think more about it.

Reply:
Using `--extra-args` sounds good, let me have a try with the script and update the README correspondingly.

Reply:
I don't think we want to use `--extra-args` directly - that will place the options after the `---`, which means they'll be retained on the *installed* system. We probably need something new/similar that can place the args *before* the `---`, e.g. a `--autoinstall-url` parameter

Reply:
Right, thanks for the reminder! I have implemented `--autoinstall-url`, and will soon update this pull request draft after making sure I have fixed according to all suggestion and discussion and rebased.